x/market/handler/endblock_test.go

```diff
@@ -0,0 +1,126 @@
+package handler
+
+import (
+	"errors"
+	"math/rand"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/ovrclk/akash/testutil"
+	"github.com/ovrclk/akash/x/market/types"
+)
+
+var (
+	// Create and seed the generator.
+	// Typically a non-fixed seed should be used, such as time.Now().UnixNano().
+	// Using a fixed seed will produce the same output on every run.
+	r = rand.New(rand.NewSource(550000))
+)
+
+type winnerTest struct {
+	desc      string
+	bids      []types.Bid
+	expWinner *types.Bid
+	expErr    error
+}
+
+func (w *winnerTest) testFunc(t *testing.T) {
+	winner, err := pickBidWinner(w.bids)
+	if !errors.Is(err, w.expErr) {
+		t.Errorf("returned err: %v does not match %v", err, w.expErr)
+	}
+	if w.expWinner != nil && !winner.Equals(w.expWinner.ID()) {
+		t.Errorf("unexpected winner: %#v\n%q : %v", winner, types.BidIDString(winner.BidID), winner.Price)
+		t.Logf("winner: %+v coin: %s", winner.ID(), winner.Price.String())
+	}
+}
+
+func TestBidWinner(t *testing.T) {
+	ownerAddr := testutil.AccAddress(t)
+	originOID := types.OrderID{
+		Owner: ownerAddr,
+		DSeq:  uint64(20),
+		GSeq:  uint32(1),
+		OSeq:  uint32(1),
+	}
```

Comment:
Please use the generator - it makes modifying things easier in the future and removes the burden of making sure that all fields are populated and populated correctly: https://github.com/ovrclk/akash/blob/b2752f0d280bd5f91ba6f27b9b263588d5c8c29e/testutil/ids.go#L40-L43

Reply:
good call!